### DISCUSSION TITLE
Use external tables instead of Directus internals

### DISCUSSION THREAD

**Discussion by awesome-developer-20** [directus/directus#11067] [Feature Requests]
## Feature Request:
Use external tables instead of Directus internals. For example, instead of using `directus_users` use `users` for authentication or instead of `directus_files` use `media` on users database.
___
## Original post:
My client already has users roles database setup for his organisation in a weird and unconventional way and they want to use that same data for directus portal login.

I've wrote a separate API for authentication which takes email and password in request and returns it's roles if user is successfully authenticated and null if not.

Is there a way where instead of duplicating the data in directus_users table, I could use the above mentioned API?

**Comment by joselcvarela** [comment]
Hello @awesome-developer-20 
From what I can tell, every login will need to have a record in `directus_users` as you can see here:
https://github.com/directus/directus/blob/86b7f7f6847a3edbd0dc68cb0f439f086938ef0e/api/src/services/authentication.ts#L60-L80

So I am afraid your scenario will not be possible.
Depending on what you are trying to achieve, it may be possible to create a single user on Directus to requests data then have a custom authentication logic so users can have access to this user or not.

**Reply by awesome-developer-20** [reply]
Thanks a lot for your quick reply 🙂 

I have thought of doing that but I need username of the logged in user.

I've a frontend application that uses directus sdk to login where users can raise an issue. So, I need to map user that raised issue with the issue.

**Reply by awesome-developer-20** [reply]
Ok no worries. I'll try to implement it someway but it would be good to see something like this in future where I could store users and roles my way and somehow use directus auth app component and api with it.

Thanks for your help and support 👍

**Reply by awesome-developer-20** [reply]
If possible mark this as potential future feature of directus 🙂 
EDIT: I've moved this discussion from "Q&A" to "Ideas" 

Something like https://github.com/directus/directus/discussions/10141#discussioncomment-1718048

**Comment by joselcvarela** [comment]
>Keywords:
external table directus own user mapping authentication files directus_users directus_files

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
